app/photo.py

```diff
@@ -71,6 +72,29 @@ def create_photo(image, handler):
     photo_url = get_photo_url(photo, handler)
     return (photo, photo_url)
 
+
+def set_thumbnail(photo):
+    image = images.Image(photo.image_data)
+    if max(image.width, image.height) <= MAX_THUMBNAIL_DIMENSION:
+        # Don't need a thumbnail, it's small enough already.
+        return
+    elif image.width > image.height:
+        image.resize(MAX_THUMBNAIL_DIMENSION,
+                     image.height * MAX_THUMBNAIL_DIMENSION / image.width)
+    else:
+        image.resize(image.width * MAX_THUMBNAIL_DIMENSION / image.height,
+                     MAX_THUMBNAIL_DIMENSION)
+    try:
+        thumbnail_data = image.execute_transforms(output_encoding=images.PNG)
+    except RequestTooLargeError:
+        raise SizeTooLargeError()
+    except Exception:
```

Comment:
Any specific exceptions you want to catch here? Would be better to catch more specific exception types here rather than catching all Exception.

Reply:
This is based on the general resizing function above, which catches Exception. It looks like there's a variety of images package and other errors that can come up so they decided to catch everything to be safe. I've copied the comment there down to my function too.